src/Route/RouteDefinitionFactory.php

```diff
@@ -0,0 +1,56 @@
+<?php declare(strict_types = 1);
+
+namespace BrandEmbassy\Slim\Route;
+
+use BrandEmbassy\Slim\DI\ServiceProvider;
+use BrandEmbassy\Slim\Middleware\MiddlewareFactory;
+use LogicException;
+use Nette\DI\Container;
+use function array_map;
+
+final class RouteDefinitionFactory
+{
+    /**
+     * @var Container
+     */
+    private $container;
+
+    /**
+     * @var MiddlewareFactory
+     */
+    private $middlewareFactory;
+
+
+    public function __construct(Container $container, MiddlewareFactory $middlewareFactory)
+    {
+        $this->container = $container;
+        $this->middlewareFactory = $middlewareFactory;
+    }
+
+
+    /**
+     * @param array<string, mixed> $routeDefinitionData
+     */
+    public function create(string $method, array $routeDefinitionData): RouteDefinition
+    {
+        $route = $this->getRoute($routeDefinitionData[RouteDefinition::SERVICE]);
+        $middlewares = array_map(
+            [$this->middlewareFactory, 'createFromConfig'],
```

Comment:
I don't like this identifying method to call by string, it can be easily forgotten when renaming the method. Can we rewrite it to classical method calling?